cg/services/analysis_starter/configurator/implementations/raredisease.py

```diff
@@ -1,95 +1,55 @@
-            params_file=self._get_params_file_path(case_id=case_id).as_posix(),
-            work_dir=self._get_work_dir(case_id=case_id).as_posix(),
+    def __init__(
+        self,
+        config: RarediseaseConfig,
+        store: Store,
+        housekeeper_api: HousekeeperAPI,
+        lims: LimsAPI,
+    ):
+        super().__init__(config=config, store=store, housekeeper_api=housekeeper_api, lims=lims)
+        self.sample_sheet_content_creator = RarediseaseSampleSheetContentCreator(
+            store=self.store, housekeeper_api=self.housekeeper_api, lims=self.lims
         )
-
-    def _create_case_directory(self, case_id: str, dry_run: bool = False) -> None:
-        """Create case working directory."""
-        case_path: Path = self._get_case_path(case_id=case_id)
-        if dry_run:
-            LOG.info(f"Would have created case directory {case_path.as_posix()}")
-            return
-        case_path.mkdir(parents=True, exist_ok=True)
-        LOG.debug(f"Created case directory {case_path.as_posix()} successfully")
-
-    def _create_nextflow_config(self, case_id: str, dry_run: bool = False) -> None:
-        """Create nextflow config file."""
-        content: str = self._get_nextflow_config_content(case_id=case_id)
-        file_path: Path = self._get_nextflow_config_path(case_id=case_id)
-        write_content_to_file_or_stdout(content=content, file_path=file_path, dry_run=dry_run)
-        LOG.debug(f"Created nextflow config file {file_path.as_posix()} successfully")
-
-    def _get_case_path(self, case_id: str) -> Path:
-        """Path to case working directory."""
-        return Path(self.root_dir, case_id)
-
-    def _get_case_priority(self, case_id: str) -> Priority:
-        return self.store.get_case_by_internal_id(case_id).priority
-
-    def _get_case_workflow(self, case_id: str) -> Workflow:
-        case: Case = self.store.get_case_by_internal_id(case_id)
-        return Workflow(case.data_analysis)
-
-    def _get_nextflow_config_content(self, case_id: str) -> str:
-        config_files_list: list[str] = [
-            self.platform,
-            self.workflow_config_path,
-            self.resources,
-        ]
-        case_specific_params: list[str] = [
-            self._get_cluster_options(case_id=case_id),
-        ]
-        return concat_txt(
-            file_paths=config_files_list,
-            str_content=case_specific_params,
+        self.params_content_creator = RarediseaseParamsFileContentCreator(
+            store=self.store, lims=self.lims, params=config.params
         )
 
-    def _get_nextflow_config_path(self, case_id: str) -> Path:
-        return Path((self._get_case_path(case_id)), f"{case_id}_nextflow_config").with_suffix(
-            FileExtensions.JSON
-        )
+    def _do_pipeline_specific_actions(self, case_id: str) -> None:
+        """Perform pipeline specific actions."""
+        self._create_gene_panel(case_id)
```

Comment:
Raredisease requires two additional files to be created wrt the other Nextflow pipelines. The implementation of the file content will be done soon, following the FileContentCreator pattern